### ISSUE TITLE
Order of evaluation causes file-not-found in `save_html`

### ISSUE THREAD

**Issue by kcf-jackson** [rstudio/htmltools#106]
In the source code of `save_html` (below), the `setwd` call in line 3 seems to affect the `html` in line 5 such that if `html` is constructed on the fly, it may be on the wrong path. 
```
function (html, file, background = "white", libdir = "lib") {
  dir <- dirname(file)
  oldwd <- setwd(dir)
  on.exit(setwd(oldwd), add = TRUE)
  rendered <- renderTags(html)
  ...
}
```

---

Here is an example:

**some_content.md**
``` 
I am a div.
```

**main.R**
```
my_html <- function() {
  x <- readLines("some_content.md", n = 1)
  htmltools::tags$div(x) 
}
x <- tempfile()

htmltools::save_html(my_html(), file = x)
# Error in file(con, "r") : cannot open the connection
# In addition: Warning message:
#   In file(con, "r") :
#   cannot open file 'some_content.md': No such file or directory
# Called from: file(con, "r")
# Browse[1]> Q

y <- my_html()
htmltools::save_html(y, file = x)
# Runs fine
```

Rendering the tags before the change of path seems to fix it.

**Comment by greg-minshall** [comment]
(sorry to pop in...)  isn't this fixed by [8ae7ee1fc742aef1dac835310b595c7e90c20cd5](https://github.com/rstudio/htmltools/commit/8ae7ee1fc742aef1dac835310b595c7e90c20cd5)?

**Comment by cpsievert** [comment]
I believe it is (please reopen if not)
